### ISSUE TITLE
Shell flyout background color does not change when behavior is set to Locked

### ISSUE THREAD

**Issue by Fabi** [dotnet/maui#6552]
### Description

The Shell Flyout background color won't change when the FlyoutBehavior is set to Locked. This worked like 4 previews ago but is broken since then. It works when using FlyoutBehavior = Flyout though.

### Steps to Reproduce

1. Create a new .NET MAUI project.
2. use this snippet in your AppShell.xaml:
```xml
<?xml version="1.0" encoding="UTF-8" ?>
<Shell
    x:Class="SampleApp.AppShell"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:local="clr-namespace:SampleApp"
    FlyoutBehavior="Locked"
    FlyoutBackground="Red"
    >

    <ShellContent Title="Home" Icon="&#xE80F;" Route="MainPage" ContentTemplate="{DataTemplate local:MainPage}" />
    <ShellContent  Title="Store" Icon="&#xE719;" Route="MainPage" ContentTemplate="{DataTemplate local:MainPage}" />
    <ShellContent Title="Library" Icon="&#xE8F1;" Route="MainPage" ContentTemplate="{DataTemplate local:MainPage}" />


</Shell>
```
3. See the background of the flyout always uses the dark/light theme default.

### Version with bug

Release Candidate 2 (current)

### Last version that worked well

Preview 12

### Affected platforms

Windows, I was *not* able test on other platforms

### Affected platform versions

All supported Windows SDK versions.

### Did you find any workaround?

For a locke flyout, no.

**Comment by kristinx0211** [comment]
can be verified this on windows, repro project:
[MauiApp3.zip](https://github.com/dotnet/maui/files/8573626/MauiApp3.zip)

**Comment by Fabi** [comment]
A comment on another issue lead to kind of a workaround that makes it working, however it really only works when placing like 2 second delay. 

https://github.com/dotnet/maui/issues/5808#issuecomment-1114181337

**Comment by Redth** [comment]
This seems related to #6701

**Comment by rachelkang** [comment]
Fixed by #6901!
